models/Note.js

```diff
@@ -116,9 +118,22 @@ class Note extends BaseModel {
 
   static async delete (shortId /*: string */) /*: Promise<any> */ {
     const noteId = translator.toUUID(shortId)
-    return await Note.query().patchAndFetchById(noteId, {
-      deleted: new Date().toISOString()
-    })
+    let note = await Note.query().findById(noteId)
+    if (!note) return null
+    note.deleted = new Date().toISOString()
+    return await Note.query().updateAndFetchById(noteId, note)
+  }
+
+  static async update (object /*: any */) /*: Promise<any> */ {
+    // $FlowFixMe
+    const noteId = translator.toUUID(parseurl(object.id).path.substr(6))
+    const modifications = _.pick(object, ['content', 'oa:hasSelector'])
```

Comment:
I assume we wont need to modify anything other than content and oa:hasSelector . Or maybe just content?

Reply:
Content definitely. oa:hasSelector is a maybe.

I can definitely think of potential scenarios in the future where you'd modify the selector. Like somebody dragging an annotation from once place in the document to another. But that's a hypothetical feature far far far down the line so we'd have plenty of time to update the api server to get it working if we decide to do it.